walkthrough/06-includes-vs-stacks/04-combining-patterns/README.md

```diff
@@ -0,0 +1,164 @@
+
+Personally, I would actually prefer less-DRY configurations, and have `api` defined independently from `rng-api` with no `_envcommon`, but different teams have different needs. You have to decide where the correct trade-offs are for the infrastructure that you are managing.
+
+Also note that the "Stack" abstraction has changed here. Instead of defining _all_ the infrastructure that lives in `live` using one `terragrunt.stack.hcl` file, multiple `terragrunt.stack.hcl` files have been used in each environment (`live/dev/terragrunt.stack.hcl` and `live/prod/terragrunt.stack.hcl`). Teams with a large IaC footprint probably already have many `terragrunt.hcl` files committed to their repositories, and tooling that interacts with the directories where those configurations are defined (e.g. in CI, you might navigate to `live/prod` and apply everything there when you merge to `main`). Rather than retool your CI/CD to leverage the new `terragrunt.stack.hcl` file, you can simply keep your existing tooling, and define your stacks the way that makes sense for your team.
+
+Additionally, note the introduction of the `environment.yml` file in the environment stacks. Many teams today today use non-HCL configurations in conjunction with HCL configurations to define configurations for their environments. The `values` attribute of the `terragrunt.stack.hcl` file is useful, and many teams might prefer it to an `environment.hcl` file found in a parent folder. However, if you are a team that is already using configuration files like `environment.yml`, and have additional tooling like `yq` that interacts with those files, you might prefer to continue using those files. This is a good example of how the Stacks pattern really aims to be additive to your existing workflows and patterns, rather than a complete replacement. You don't have to rewrite all of your configurations if it doesn't benefit you.
+
+## Conclusion
+
+Hopefully, these examples give you a good idea of how you can use the different patterns enabled by Terragrunt to shape configurations in a way that works for your team, and the trade-offs they present.
+
+The "Not DRY" pattern is a good way to get started with Terragrunt, and is a good way to get a feel for how Terragrunt works. It's also a good way to get started with Terragrunt if you're not sure how you want to structure your configurations yet. However, as your infrastructure grows, you'll likely find that you want to share more configuration between units, and the "Not DRY" pattern can become unwieldy.
+
+The "_envcommon" pattern is a good way to quickly configuration between units without larger refactoring, and is a valid way to get started with reusing configurations. However, as the number of units you manage grows, you'll likely find that you want to share more configuration between environments, and the "_envcommon" pattern can become unwieldy.
```

Comment:
```suggestion
The "_envcommon" pattern is a good way to quickly share configuration between units without larger refactoring, and is a valid way to get started with reusing configurations. However, as the number of units you manage grows, you'll likely find that you want to share more configuration between environments, and the "_envcommon" pattern can become unwieldy.
```

In addition to the typo - I think this summary paragraph is missing the key insight as to why `_envcommon` is unweildy.  It:
* It doesn't provide for flexible configuration between instantiations
* Requires the local filesystem, so all callers get the same version and thus its challenging to roll out to multiple environments progressively